crates/ty_python_semantic/src/types.rs

```diff
@@ -899,6 +899,22 @@ impl<'db> Type<'db> {
         matches!(self, Type::Dynamic(_))
     }
 
+    /// Is a value of this type only usable in typing contexts?
+    pub(crate) fn is_type_check_only(&self, db: &'db dyn Db) -> bool {
+        match self {
+            Type::ClassLiteral(class_literal) => class_literal.type_check_only(db),
+            Type::FunctionLiteral(f) => {
+                if f.is_known(db, KnownFunction::TypeCheckOnly) {
```

Comment:
If there a way to avoid this somehow? Maybe `typing.type_check_only` could get the `TYPE_CHECK_ONLY` decorator flag when it's "inserted" into the typing module? I wasn't able to find where that happens